R/tune_bayes.R

```diff
@@ -494,6 +498,16 @@ create_initial_set <- function(param, n = NULL, checks) {
   dials::grid_latin_hypercube(param, size = n)
 }
 
+check_iter <- function(iter, call) {
+  if (!(is.numeric(iter) && length(iter) == 1L && !is.na(iter) && iter >= 0)) {
```

Comment:
Modeled after `isTRUE` but for a single integer, in favor of porting over the rlang checkers.

Reply:
Considered revisiting this to rewrite with `val_class_and_single()`, but the predicate logic there wouldn't accept `1` rather than `1L` as input, and the function doesn't accept a proper `call`ing env.